src/main/java/team/asd/controller/ProductController.java

```diff
@@ -21,37 +22,21 @@ public ProductController(@Autowired ProductService productService) {
 
     @GetMapping(path = {"/{id}"})
     public ResponseEntity<Object> getProductById(@PathVariable Integer id) {
-        try {
-            return new ResponseEntity<>(ConverterUtil.convertProductToDto(productService.readById(id)), HttpStatus.OK);
-        } catch (ValidationException e) {
-            return new ResponseEntity<>(e.getMessage(), HttpStatus.BAD_REQUEST);
-        }
+        return new ResponseEntity<>(ConverterUtil.convertProductToDto(productService.readById(id)), HttpStatus.OK);
```

Comment:
You don’t need to wrap the DTO in a ResponseEntity, since you use the @RestController annotation, I would also recommend returning the DTO and not the Object in the method.